### DISCUSSION TITLE
Support multiple return values

### DISCUSSION THREAD

**Discussion by kunitoki** [kunitoki/LuaBridge3#74] [Ideas]
Sometimes you might want to push multiple values for a type.

For example when pushing all values of a variant type or std::pair. Currently it's done by pushing them in a table.

But in my case I don't want a table to return, I want each value separately in Lua.

```C++
[[nodiscard]] static Result push(lua_State* L, Type const& e)
{
	// push a value
	// push another value
	Result r;
	r.num_args(2); // We pushed two values so set it to 2
	return r;
}
```

**Comment by kunitoki** [comment]
This means we might need to change `Stack`, but it's difficult to come up with a solution that could work without making thing super complicated and unflexible, especially if we push the burden of supporting both modes to the developer integrating its own types.

Also, if you need this from lua, you could use `table.unpack` like:
```lua
local array = {1, 2, 3, 4}
local a, b, c, d = table.unpack(array)
```

**Comment by rpatters1** [comment]
Is there some way to use `std::tie` or `std::tuplet`? I have often wished there was some way to have a C function return a `std::tuplet` (or `pair`) and have LB spin it out on the stack as separate return values. Maybe an option when the function is registered? Or a wrapper class that does the same?
